src/main/java/uk/gov/hmcts/reform/propertiesvolume/PropertiesvolumePropertySource.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.hmcts.reform.propertiesvolume;
+
+import org.springframework.core.env.MapPropertySource;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+
+/**
+ * Note: this class reuses code from spring-cloud-kubernetes
+ * (https://github.com/spring-cloud/spring-cloud-kubernetes).
+ */
+public class PropertiesvolumePropertySource extends MapPropertySource {
```

Comment:
looks like you need to add something like 
the addToEnvironment bit in the RandomPropertySource
https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/env/RandomValuePropertySource.java

Reply:
Why would I need to do that?